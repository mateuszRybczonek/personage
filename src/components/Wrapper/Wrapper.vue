<template>
  <div :class="$style.wrapperContainer">
    <nav
      v-if="!noHeader"
      :class="$style.nav"
      data-test="header"
    >
      <button
        v-if="back"
        data-test="wrapper-go-back-arrow"
        :class="$style.navLink"
        :aria-label="$t('general.go_back')"
        @click="$_redirect(back)"
      >
        <BackIcon :class="$style.icon" />
      </button>
      <div :class="$style.navTitle">
        <slot name="header" />
      </div>
      <button
        v-if="close"
        :class="[
          $style.navLink,
          $style.pullRight,
        ]"
        :aria-label="$t('general.close')"
        data-test="close-page"
        @click="$_redirect(close)"
      >
        <CloseIcon :class="$style.icon" />
      </button>
    </nav>
    <main
      :class="[$style.wrapper, {
        'no-header': noHeader,
        'no-padding': noPadding
      }]"
    >
      <slot />
      <footer
        data-test="footer"
        :class="$style.wrapperFooter"
      >
        <slot name="footer" />
      </footer>
    </main>
  </div>
</template>

<script>
import BackIcon from '@/assets/back.svg';
import CloseIcon from '@/assets/close.svg';

export default {
  components: {
    BackIcon,
    CloseIcon,
  },

  props: {
    back: {
      type: Object,
      default: null,
    },
    close: {
      type: Object,
      default: null,
    },
    noHeader: {
      type: Boolean,
      default: false,
    },
    noPadding: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style module lang="scss">
  $headerHeight: 60px;

  .nav {
    z-index: $z-index-1;
    display: flex;
    height: rem($headerHeight);
    background-color: $c-background;
    transform: translateZ(0);
    align-items: center;
    justify-content: space-between;
  }

  .navTitle {
    @include absolute(right 0 left 0);

    font-size: $fs-h3;
    font-weight: $fw-bold;
    line-height: rem($headerHeight);
    text-align: center;
    color: $c-heading;
  }

  .navLink {
    @include sizing(52px 60px);

    position: relative;
    z-index: $z-index-1;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .icon {
    @include sizing(15px);

    display: inline-block;
    color: $c-white;
  }

  .wrapperContainer {
    position: relative;
    height: 100%;
  }

  .wrapper {
    @include absolute(top $headerHeight right 0 left 0 bottom 0);

    display: flex;
    overflow-y: scroll;
    padding: 0 rem(30px) rem(30px) rem(30px);
    flex-direction: column;
    -webkit-overflow-scrolling: touch;

    &:global(.no-header) {
      top: 0;
    }

    &:global(.no-padding) {
      padding-right: 0;
      padding-left: 0;
    }
  }

  .wrapperFooter {
    margin-top: auto; /* pull down */
    padding-top: 20px;
    font-size: $fs-button-sm;

    @media #{$mobile-lg-v-up} {
      padding-top: 30px;
      font-size: $fs-button;
    }

    .wrapper:global(.no-padding) & {
      padding-right: 30px;
      padding-left: 30px;
    }

    :global(.btnPrimary) {
      margin: auto auto rem(12px);

      @media #{$mobile-md-v-up} {
        margin-bottom: rem(26px);
      }
    }
  }

  .pullRight {
    margin-left: auto;
  }
</style>
