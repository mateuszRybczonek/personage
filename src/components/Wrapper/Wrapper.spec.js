import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Wrapper from './Wrapper';

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
const router = new VueRouter();

const URL = '/url';
const $t = key => key;

describe('Wrapper', () => {
  it('renders title', () => {
    const wrapper = shallowMount(Wrapper, {
      slots: {
        header: 'This is a title.',
        footer: '<div>footer</div>',
        default: '<div>content</div>',
      },
      mocks: {
        $t,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('doesn\'t render header if "noHeader" props is passed', () => {
    const wrapper = shallowMount(Wrapper, {
      propsData: {
        noHeader: true,
      },
      mocks: {
        $t,
      },
    });
    expect(wrapper.findAll('[data-test="header"]')).toHaveLength(0);
  });

  it('renders back link if url is given', () => {
    const wrapper = mount(Wrapper, {
      localVue,
      router,
      propsData: {
        back: { path: URL },
      },
      mocks: {
        $t,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders close link if url is given', () => {
    const wrapper = mount(Wrapper, {
      localVue,
      router,
      propsData: {
        close: { path: URL },
      },
      mocks: {
        $t,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  context('language flag', () => {
    const store = new Vuex.Store({
      modules: {
        settings: {
          namespaced: true,
          state: {
            locale: 'en',
          },
        },
      },
    });

    const setup = (showLocale) => {
      const wrapper = shallowMount(Wrapper, {
        localVue,
        store,
        propsData: {
          showLocale,
        },
        mocks: {
          $t,
        },
      });

      return { wrapper };
    };

    it('is rendered when showLocale is true', () => {
      const { wrapper } = setup(true);
      expect(wrapper.findAll('[data-test="language-flag"]')).toHaveLength(1);
    });

    it('is not rendered when showLocale is false', () => {
      const { wrapper } = setup(false);
      expect(wrapper.findAll('[data-test="language-flag"]')).toHaveLength(0);
    });
  });
});
