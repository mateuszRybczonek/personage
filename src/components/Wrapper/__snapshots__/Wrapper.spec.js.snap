// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper renders back link if url is given 1`] = `
<div>
  <nav
    data-test="header"
  >
    <button
      aria-label="general.go_back"
      data-test="wrapper-go-back-arrow"
    >
      <svg
        height="15"
        viewBox="0 0 10 15"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 14L1 7.5 9 1"
          fill="none"
          fill-rule="evenodd"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
     
    <div />
     
    <!---->
     
    <!---->
  </nav>
   
  <main
    class=""
  >
     
    <footer
      data-test="footer"
    />
  </main>
</div>
`;

exports[`Wrapper renders close link if url is given 1`] = `
<div>
  <nav
    data-test="header"
  >
    <!---->
     
    <div />
     
    <!---->
     
    <button
      aria-label="general.close"
      class=""
      data-test="close-page"
    >
      <svg
        height="14"
        viewBox="0 0 14 14"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.7.3c-.4-.4-1-.4-1.4 0L7 5.6 1.7.3C1.3-.1.7-.1.3.3c-.4.4-.4 1 0 1.4L5.6 7 .3 12.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3.3 0 .5-.1.7-.3L7 8.4l5.3 5.3c.2.2.5.3.7.3.2 0 .5-.1.7-.3.4-.4.4-1 0-1.4L8.4 7l5.3-5.3c.4-.4.4-1 0-1.4z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </button>
  </nav>
   
  <main
    class=""
  >
     
    <footer
      data-test="footer"
    />
  </main>
</div>
`;

exports[`Wrapper renders title 1`] = `
<div>
  <nav
    data-test="header"
  >
    <!---->
     
    <div />
     
    <!---->
     
    <!---->
  </nav>
   
  <main
    class=""
  >
    This is a title.
    <div>
      content
    </div>
     
    <footer
      data-test="footer"
    >
      <div>
        footer
      </div>
    </footer>
  </main>
</div>
`;
